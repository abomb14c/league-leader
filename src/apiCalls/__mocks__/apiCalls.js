import { cleanEPLData } from '../../mockData/mockEPLData';
import { mockCleanNBAData } from '../../mockData/mockNBAData';


export const fetchEnglandScores = jest.fn().mockImplementation(() => 
  Promise.resolve(cleanEPLData));


export const fetchNbaTeams = jest.fn().mockImplementation(() => 
  Promise.resolve(mockCleanNBAData));

export const addUserFetch = jest.fn().mockImplementation(() => Promise.resolve({
  "userId": 1
}));

export const fetchUser = jest.fn().mockImplementation(() => Promise.resolve({
  data: {
    id: 2,
    name: 'alan',
    password: 'password',
    email: 'alanjcharles14@gmail.com'
  }
}));