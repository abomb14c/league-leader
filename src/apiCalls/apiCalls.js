import  soccerKey  from '../apiKeys';
import { cleanEnglishSoccer } from '../cleaner/cleaner';

export const fetchEnglandScores = async () => {
  const url = `https://api.sportradar.us/soccer-t3/eu/en/tournaments/sr:tournament:17/live_standings.json?api_key=${soccerKey}`
  const response = await fetch(url);
  const dirtySoccerData = await response.json();
  const soccerData = dirtySoccerData.standings[0].groups[0].team_standings;

 return cleanEnglishSoccer(soccerData);
}

