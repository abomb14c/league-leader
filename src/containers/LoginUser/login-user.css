.login-button {
  border: none;
  background-color: #7E8F7C;
  color: #fff;
  font-size: 14px;
  width: 84%;
  height: 35px;
}

.login-button:hover {
  background-color: #C63D0F;
}

.login-form {
  display: flex;
  flex-direction: column;
}

.login-user {
  height: 30px;
  margin-bottom: 5%;
  padding-left: 2%;
  width: 80%;
}

.login-title {
  color: #C63D0F;
  font-size: 28px;
}