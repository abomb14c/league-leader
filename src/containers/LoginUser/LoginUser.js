import React, { Component } from 'react'
import './login-user.css'

export class LoginUser extends Component {
  constructor(props){
    super(props)

    this.state = {
      email: '',
      password: ''
    }
  }

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };
  
  render() {
    return (
      <form className="login-form">
        <h3 className="login-title">Login</h3>
        <input
          className="login-user"
          type="text"
          name="email"
          value={this.state.email}
          placeholder="Email"
          onChange={this.handleChange}
        />
        <input
          className="login-user"
          type="password"
          name="password"
          value={this.state.password}
          placeholder="Password"
          onChange={this.handleChange}
        />
        <button className="login-button">Login</button>
      </form>
    )
  }
}

export default LoginUser;