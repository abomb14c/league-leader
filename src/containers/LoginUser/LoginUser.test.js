import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import  { LoginUser, mapDispatchToProps}  from './LoginUser';
import { fetchUser } from '../../apiCalls/apiCalls';

// jest.mock('./../../apiCalls/apiCalls');

describe('DraftCard', () => {
  let wrapper;
  let mockHandleLogin;

  beforeEach(() => {
    mockHandleLogin = jest.fn();

    wrapper = shallow(<LoginUser 
      handleLogin={mockHandleLogin}
    />);
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has a default state", () => {
    let expected = {
      username: "",
      password: ""
    };

    expect(wrapper.state()).toEqual(expected);
  });


});