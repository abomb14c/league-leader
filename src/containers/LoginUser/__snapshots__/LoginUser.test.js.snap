// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginUser
    handleLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="login-title"
>
          Login
</h3>,
        <input
          className="login-user"
          name="username"
          onChange={[Function]}
          placeholder="username"
          type="text"
          value=""
/>,
        <input
          className="login-user"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
/>,
        <button
          className="login-button"
>
          Login
</button>,
      ],
      "className": "login-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "className": "login-title",
        },
        "ref": null,
        "rendered": "Login",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login-user",
          "name": "username",
          "onChange": [Function],
          "placeholder": "username",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login-user",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "className": "login-button",
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="login-title"
>
            Login
</h3>,
          <input
            className="login-user"
            name="username"
            onChange={[Function]}
            placeholder="username"
            type="text"
            value=""
/>,
          <input
            className="login-user"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
          <button
            className="login-button"
>
            Login
</button>,
        ],
        "className": "login-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "login-title",
          },
          "ref": null,
          "rendered": "Login",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login-user",
            "name": "username",
            "onChange": [Function],
            "placeholder": "username",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login-user",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "login-button",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
